src/day_6.rs

```diff
@@ -0,0 +1,50 @@
+use std::fs::File;
+use std::io::{self};
+
+fn get_timers(mut lines: io::Lines<io::BufReader<File>>, epoch: u16) -> u64 {
+  let mut timers = lines
+    .next()
+    .unwrap()
+    .unwrap()
+    .split(",")
+    .fold(vec![0; 10], |mut accum, s| {
+      if let Ok(timer) = s.parse::<usize>() {
+        accum[timer] += 1;
+      }
+
+      accum
+    });
+
+  for _ in 0..epoch {
+    let mut updates: Vec<i64> = vec![0; 10];
+
+    // Iterate backwards so new items only
+    // decrement on next loop.
+    for i in (0..timers.len()).rev() {
+      if i == 0 {
+        updates[8] += timers[i];
+        updates[6] += timers[i];
+        updates[0] -= timers[i];
+      } else {
+        if timers[i] > 0 {
+          updates[i] -= timers[i];
+          updates[i - 1] += timers[i];
+        }
+      }
+    }
```

Comment:
is doable 'in place' with .rotate_left/rotate_right and adding extra to the one that needs extra (all super adults become babies, but they also become teenagers)

Reply:
love it - way nicer with `.rotate_left` https://github.com/jling90/advent-of-code-2021/commit/d5aecaf915e7c3b85295b361628a4b0ba43003e6

Reply:
https://github.com/RyanCarrier/advent-of-code-2021/blob/main/src/day6.rs#L7-L11

this was my 'step' function